utils/hct/hctdb.py

```diff
@@ -8081,6 +8081,56 @@ def build_valrules(self):
             "%0 node shader '%1' has incompatible input record type (should be %2)",
         )
 
+        # These errors are emitted from ShaderCompatInfo validation.
+        # If a called function is identifiable as a potential source of the
+        # incompatibility, you get Sm.IncompatibleCallInEntry,
+        # otherwise you get Sm.IncompatibleOperation.
+        # You also get the specific incompatibilities found with one function
+        # introducing each problem.
+        # These may be emitted in addition to another specific operation
+        # validation error that identifies the root cause, but is meant to
+        # catch cases currently missed by other validation.
+        self.add_valrule_msg(
+            "Sm.IncompatibleCallInEntry",
+            "Features used in internal function calls must be compatible with entry",
+            "Entry function calls one or more functions using incompatible features.  See other errors for details.",
+        )
+        self.add_valrule_msg(
+            "Sm.IncompatibleOperation",
+            "Operations used in entry function must be compatible with shader stage and other properties",
+            "Entry function performs some operation that is incompatible with the shader stage or other entry properties.  See other errors for details.",
+        )
+        self.add_valrule_msg(
+            "Sm.IncompatibleStage",
+            "functions may only use features available in the entry function's stage",
+            "Function uses features incompatible with the shader stage (%0) of the entry function.",
+        )
+        self.add_valrule_msg(
+            "Sm.IncompatibleShaderModel",
+            "functions may only use features available in the current shader model",
+            "Function uses features incompatible with the shader model.",
+        )
+        self.add_valrule_msg(
+            "Sm.IncompatibleThreadGroupDim",
+            "When derivatives are used in compute-model shaders, the thread group dimensions must be compatible",
+            "Function uses derivatives in compute-model shader with NumThreads (%0, %1, %2); derivatives require NumThreads to be 1D and a multiple of 4, or 2D/3D with X and Y both being a multiple of 2.",
```

Comment:
Looks like there isn't a test for this diagnostic.

Reply:
Good catch, working on these.